### PR TITLE
U/sschmidt23/mwdust

### PR DESCRIPTION
**Pull request by sschmidt23**
This PR adds a simple dust dereddening notebook for dc2 object catalogs, addressing the point in issue #121.  I deleted the previous version of the notebook that was more complicated and relied on lsst.sims.photUtils, this notebook uses the dustmaps package that Heather added to desc-python-dev.

### CODE REVIEW

File: contributed/DC2_object_deredden.ipynb [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import GCRCatalogs\n",
+    "import dustmaps\n",
+    "from dustmaps.sfd import SFDQuery\n",
+    "from astropy.coordinates import SkyCoord\n",
+    "from dustmaps.config import config\n",
+    "import matplotlib.pyplot as plt\n",
+    "config['data_dir'] = '/global/cfs/cdirs/lsst/groups/PZ/PhotoZDC2/run2.2i_dr6_test/TESTDUST/mapdata' #update this path when dustmaps are copied to a more stable location!\n",
```
Comment (yymao) [inline contributed/DC2_object_deredden.ipynb:44]: Should we move this file to the `shared` directory?

File: contributed/DC2_object_deredden.ipynb [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    "Let's grab a sample set of DC2 data to deredden, we'll use run2.2i_dr3 and use tract 3450, and store it in a pandas dataframe for simplicity:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "cat = GCRCatalogs.load_catalog(\"dc2_object_run2.2i_dr3\")"
```
Comment (yymao) [inline contributed/DC2_object_deredden.ipynb:71]: Consider updating to dr6a or dr6c?

File: contributed/DC2_object_deredden.ipynb [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    "For comparison, we have queried the truth fluxes with and without Milky Way dust from the postgres SQL database (which you can learn about in the postgres_truth.ipynb notebook in the tutorials directory).  We ran a query to grab true fluxes for stars in the same region of sky as the example tract.  To save time we've saved this data in a parquet file that we will read in with Pandas:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stardf = pd.read_parquet(\"data/demo_star_fluxes_colors_dust.parquet\",engine='pyarrow')"
```
Comment (yymao) [inline contributed/DC2_object_deredden.ipynb:215]: Should we check in the script or notebook that was used to generate this file?

**Comment by yymao** [comment]
@sschmidt23 is this branch actually merged? I couldn't find the new notebook `contributed/DC2_object_deredden.ipynb` in master branch.

**Comment by sschmidt23** [comment]
No, I think I must have accidentally merged my mwdust and extragalactic branches in my local DC2-analysis repo.  I was just going through your couple of comments to try to fix this up a bit.  Now that the truth catalog is available via gcr-catalogs I think I can switch from reading the parquet file to querying to get the stellar locus, which is a lot cleaner.  I'll push updates shortly.

File: contributed/DC2_object_deredden.ipynb (hunk 1 above)
Comment (sschmidt23) [reply contributed/DC2_object_deredden.ipynb:44]: I think so, it was still an open question as to whether to use the `dustmaps` package when I made this, but I think it's clearly a nice solution, so copying the two files to a location on shared would be best.  The files are actually pretty small, I think 2x 50 MB. 
If the path is updated I can modify the notebook to point to the new path.

File: contributed/DC2_object_deredden.ipynb (hunk 2 above)
Comment (sschmidt23) [reply contributed/DC2_object_deredden.ipynb:71]: Switched to dr6c

File: contributed/DC2_object_deredden.ipynb (hunk 3 above)
Comment (sschmidt23) [reply contributed/DC2_object_deredden.ipynb:215]: I changed the notebook to read in from GCRCatalogs star truth table, and removed the dependency on this file.

**Comment by sschmidt23** [comment]
Updated the notebook, the only potentially unresolved issue is copying the dustups data files to shared.  Otherwise I think the notebook is good to go.

I also included a separate notebook showing how to derive the A/E(B-V) coefficients for the LSST filters, this is kinda into the weeds a bit, I don't know if we need to clutter up the contributed directory with it or not.  Let me know your opinion, we can either keep it in the PR or remove, I don't have a strong opinion.

**Comment by sschmidt23** [comment]
Newest commit isn't showing up here (I think because it was closed), I'll start a new PR and reference this one so we know to re-close once it is approved.

**Comment by yymao** [comment]
Thanks @sschmidt23. 

This PR is now superseded by #146.

**Pull request merged**